lib/Sema/CSDiagnostics.cpp

```diff
@@ -388,7 +388,8 @@ ValueDecl *RequirementFailure::getDeclRef() const {
       do {
         if (auto *parent = DC->getAsDecl()) {
           if (auto *GC = parent->getAsGenericContext()) {
-            if (GC->getGenericSignature() != Signature)
+            // FIXME: Is this intending an exact match?
+            if (GC->getGenericSignature().getPointer() != Signature.getPointer())
```

Comment:
@xedin Should this canonicalize the generic signatures, or are we looking for an exact match here?